modules/avcodec/decode.c

```diff
@@ -439,7 +473,63 @@ int avcodec_decode_h264(struct viddec_state *st, struct vidframe *frame,
 		goto out;
 	}
 
-	err = ffdecode(st, frame, &pkt->intra);
+	/*	When using MediaCodec hardware decoding,
+		you must set width, height,
+		and SPS/PPS parameters before decoding the first frame,
+		otherwise decoding will fail.
+	 	Here, width, height, and extradata are set
+		by parsing the SPS/PPS.
+	 	This is only required when using MediaCodec hardware
+		decoding—other software decoders do not need this.
+	 	Additionally, the extradata format must be:
+		0x00 0x00 0x01 sps 0x00 0x00 0x01 pps */
+	if (st->mc && !st->open) {
+		uint8_t sps_data[256];
+		int sps_len;
+		uint8_t pps_data[256];
+		int pps_len;
+		int ret = h264_get_sps_pps(
+			st->mb->buf,(int)st->mb->pos,
+			sps_data, &sps_len,
+			pps_data, &pps_len);
+		if (ret) {
+			warning("avcodec: decode: "
+				"h264_get_sps_pps error %d\n",
+				ret);
+			goto out;
+		}
+		ret = h264_decode_sps_with_width_and_height(
+			sps_data,sps_len,
+			&st->ctx->width,
+			&st->ctx->height);
+		if (ret) {
+			warning("avcodec: decode: "
+				"h264_decode_sps_"
+				"with_width_and_height error %d\n",
+				ret);
+			goto out;
+		}
+		st->ctx->extradata_size = sps_len + pps_len + 6;
+		st->ctx->extradata = av_malloc(
+			st->ctx->extradata_size +
+			AV_INPUT_BUFFER_PADDING_SIZE);
+		if (!st->ctx->extradata)
+			goto out;
+		int offset = 0;
+		memcpy(st->ctx->extradata+offset, nal_seq, 3);
+		offset+=3;
+		memcpy(st->ctx->extradata+offset, sps_data, sps_len);
+		offset+=sps_len;
+		memcpy(st->ctx->extradata+offset, nal_seq, 3);
+		offset+=3;
+		memcpy(st->ctx->extradata+offset, pps_data, pps_len);
+		st->open = true;
+		debug("avcodec: decode: init decoder H264\n");
+		/* Call the initialization encoder */
```

Comment:
Wrong wording, but I think you can remove this comment completely, it's redundant.